src/main/java/backend/Logic.java

```diff
@@ -157,37 +160,84 @@ public void refresh() {
         for (Map.Entry<Integer, IssueMetadata> entry : metadata.entrySet()) {
             IssueMetadata currentMetadata = entry.getValue();
             Date lastNonSelfUpdate = new Date(0);
+            Date lastSelfUpdate = new Date(0);
```

Comment:
You should use `LocalDateTime.now()` for these. `Date` is a legacy class and could only be used for compatibility.